v7.12.0 update throw is missing in props validation  (react/prop-types)

After updating to v7.12.0 appeared strange validation errors.

I have this propTypes:
```js
propTypes = {
    actions: PropTypes.object.isRequired,
};
```

and this code:
```js
componentWillReceiveProps (nextProps) {
    this.props.actions.doSomething();
}

componentWillUnmount () {
    this.props.actions.doSomething();
}
```

And then I have eslint error:  'doSomething' is missing in props validation  react/prop-type
on the line in componentWillReceiveProps. But the line in componentWillUnmount is ok. 
Strange that it says "'doSomething' is missing" and not "'actions.doSomething' is missing"

I tried to change "PropTypes.object" to "PropTypes.shape", still the same.

Error disappears if I change name "componentWillReceiveProps" to something else. 
If I add propType  "doSomething: PropTypes.func.isRequired" error disappears too but it doesn't make sense.

after update 7.12
```    componentDidUpdate (prevProps, prevState, snapshot) {
        if (!prevProps.show && this.props.show) {
            document.body.style.overflow = 'hidden'
        } else if (prevProps.show && !this.props.show) {
            document.body.style.overflow = 'auto'
        }
    }
```

```
  43:27  error  'style' is missing in props validation           react/prop-types
  43:33  error  'overflow' is missing in props validation        react/prop-types
  43:33  error  'style.overflow' is missing in props validation  react/prop-types
  45:27  error  'style' is missing in props validation           react/prop-types
  45:33  error  'overflow' is missing in props validation        react/prop-types
  45:33  error  'style.overflow' is missing in props validation  react/prop-types

✖ 6 problems (6 errors, 0 warnings)```

--- a/lib/util/usedPropTypes.js
+++ b/lib/util/usedPropTypes.js
@@ -285,13 +285,14 @@ module.exports = function usedPropTypesInstructions(context, components, utils) 
    * @returns {Boolean} True if we are using a prop, false if not.
    */
   function isPropTypesUsage(node) {
+    const isThisPropsUsage = node.object.type === 'ThisExpression' && node.property.name === 'props';
+    const isPropsUsage = isThisPropsUsage || node.object.name === 'nextProps' || node.object.name === 'prevProps';
     const isClassUsage = (
       (utils.getParentES6Component() || utils.getParentES5Component()) &&
-      ((node.object.type === 'ThisExpression' && node.property.name === 'props')
-      || isPropArgumentInSetStateUpdater(node))
+      (isThisPropsUsage || isPropArgumentInSetStateUpdater(node))
     );
     const isStatelessFunctionUsage = node.object.name === 'props' && !isAssignmentToProp(node);
-    return isClassUsage || isStatelessFunctionUsage || inLifeCycleMethod();
+    return isClassUsage || isStatelessFunctionUsage || (isPropsUsage && inLifeCycleMethod());
   }
 
   /**

--- a/tests/lib/rules/prop-types.js
+++ b/tests/lib/rules/prop-types.js
@@ -2170,6 +2170,83 @@ ruleTester.run('prop-types', rule, {
           pragma: 'Foo'
         }
       }
+    },
+    {
+      code: `
+      class Foo extends React.Component {
+        propTypes = {
+          actions: PropTypes.object.isRequired,
+        };
+        componentWillReceiveProps (nextProps) {
+          this.props.actions.doSomething();
+        }
+
+        componentWillUnmount () {
+          this.props.actions.doSomething();
+        }
+
+        render() {
+          return <div>foo</div>;
+        }
+      }
+      `,
+      parser: 'babel-eslint'
+    },
+    {
+      code: `
+      class Foo extends React.Component {
+        componentDidUpdate() { this.inputRef.focus(); }
+        render() {
+          return (
+            <div>
+              <input ref={(node) => { this.inputRef = node; }} />
+            </div>
+          )
+        }
+      }
+      `
+    },
+    {
+      code: `
+        class Foo extends React.Component {
+          componentDidUpdate(nextProps, nextState) {
+            const {
+                first_organization,
+                second_organization,
+            } = this.state;
+            return true;
+          }
+          render() {
+            return <div>hi</div>;
+          }
+        }
+      `
+    },
+    {
+      code: `
+      class Foo extends React.Component {
+        shouldComponentUpdate(nextProps) {
+          if (this.props.search !== nextProps.search) {
+            let query = nextProps.query;
+            let result = nextProps.list.filter(item => {
+              return (item.name.toLowerCase().includes(query.trim().toLowerCase()));
+            });
+
+            this.setState({ result });
+
+            return true;
+          }
+        }
+        render() {
+          return <div>foo</div>;
+        }
+      }
+      Foo.propTypes = {
+        search: PropTypes.object,
+        list: PropTypes.array,
+        query: PropTypes.string,
+      };
+      `
     }
   ],
 

